veniq/dataset_collection/validation.py

```diff
@@ -0,0 +1,284 @@
+    end_line_SEMI: int
+    start_line_dataset: int
+    end_line_dataset: int
+    percent_matched: float
+    class_name: str
+    method_name: str
+    error_string: str
+    ncss: int
+    matched: bool
+    failed_cases_in_SEMI_algorithm: bool
+    no_opportunity_chosen: bool
+    failed_cases_in_validation_examples: bool
+
+
+def fix_start_end_lines_for_opportunity(
+        extracted_lines_of_opportunity: List[int],
+        filepath: str) -> Tuple[int, int]:
+    """
+    Finds start and end lines for opportunity
+
+    :param filepath: filename where opportunity was found
+    :param extracted_lines_of_opportunity: list of lines for opportunity
+    :return: list of extracted lines for opportunity
+    """
+    start_line_opportunity = min(extracted_lines_of_opportunity)
+    end_line_opportunity = max(extracted_lines_of_opportunity)
+    text = read_text_with_autodetected_encoding(filepath).split('\n')
+    extraction = text[start_line_opportunity:end_line_opportunity]
+    open_brackets = 0
+    close_brackets = 0
+    for x in extraction:
+        close_brackets += x.count('}')
+        open_brackets += x.count('{')
+
+    if open_brackets < close_brackets:
+        diff = close_brackets - open_brackets
+        count = 1
+        for text_line in text[end_line_opportunity:]:
+            if diff < 1:
+                break
+            else:
+                if text_line.find('{') > -1:
+                    diff -= 1
+                    count += 1
+
+        start_line_opportunity += count - 1
+
+    elif open_brackets > close_brackets:
+        diff = open_brackets - close_brackets
+        count = 1
+        for text_line in text[end_line_opportunity:]:
+            if diff < 1:
+                break
+            else:
+                if text_line.find('}') > -1:
+                    diff -= 1
+                    count += 1
+
+        end_line_opportunity += count - 1
+
+    return start_line_opportunity, end_line_opportunity
+
+
+# flake8: noqa: C901
+def validate_row(dataset_dir: Path, row: pd.Series) \
+        -> List[RowResult]:
+    """
+    Validate row of dataset
+
+    :param dataset_dir: directory to dataset, path before the relative path in
+    output_filename
+    :param row: row of dataframe
+    :return: boolean value whether we should consider this row or skip it,
+    Stats - return collected stats
+    """
+    results = []
+    try:
+        start_line_of_inserted_block = int(row[1]['inline_insertion_line_start'])
+        end_line_of_inserted_block = int(row[1]['inline_insertion_line_end'])
+
+        src_filename = row[1]['output_filename']
+        class_name = row[1]['class_name']
+        full_path = dataset_dir / src_filename
+        ast = AST.build_from_javalang(build_ast(full_path))
+        function_to_analyze = row[1]['method_where_invocation_occurred']
+
+        for class_decl in ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION):
+            if class_decl.name != class_name:
```

Comment:
```suggestion
            if class_decl.name == class_name:
                ...
```

Reply:
i will check it, seems i had some idea for doing it

Reply:
fixed